src/BenchmarkDotNet/Validators/DotNetSdkVersionValidator.cs

```diff
@@ -0,0 +1,102 @@
+﻿using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Toolchains;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class DotNetSdkVersionValidator : IValidator
+    {
+        private readonly string _customDotNetCliPath;
+        private readonly IDotNetSdkProvider _dotNetSdkProvider;
+        private readonly IDotNetFrameworkSdkProvider _dotNetFrameworkSdkProvider;
+
+        public DotNetSdkVersionValidator(string customDotNetCliPath = null)
+        {
+            _customDotNetCliPath = customDotNetCliPath;
+            _dotNetSdkProvider = new DotNetSdkProvider(_customDotNetCliPath);
+            _dotNetFrameworkSdkProvider = new DotNetFrameworkSdkProvider();
+        }
+
+        public bool TreatsWarningsAsErrors => true;
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
+        {
+            foreach (var benchmark in validationParameters.Benchmarks)
+            {
+                if (benchmark == null || benchmark.Job == null || benchmark.Job.Environment?.Runtime?.RuntimeMoniker == null)
+                {
+                    continue;
+                }
+
+                var runtimeMoniker = benchmark.Job.Environment.Runtime.RuntimeMoniker;
+                if (!IsSdkInstalled(runtimeMoniker))
+                {
+                    yield return new ValidationError(
+                        TreatsWarningsAsErrors,
+                        $"The required SDK for {runtimeMoniker} is not installed",
+                        benchmark);
+                }
+            }
+        }
+
+        private bool IsSdkInstalled(RuntimeMoniker runtimeMoniker)
+        {
+            string requiredSdkVersion = GetSdkVersionFromMoniker(runtimeMoniker);
+
+            // If the required SDK version is for .NET Framework
+            if (requiredSdkVersion.StartsWith("4"))
```

Comment:
Just make a `public static IEnumerable<ValidationError> ValidateFrameworkSdks()` and `public static IEnumerable<ValidationError> ValidateCoreSdks(string? customDotNetCliPath)` that the toolchains can call.

Reply:
ok great idea but concerning the issue in #2199, we want to verify that the SDK specified with `RuntimeMoniker` is installed. So are you ok if I add `BenchmarkCase benchmarkCase` as an additional parameter in these static methods? This will allow me access to `benchmarkCase.Job.Environment.Runtime?.RuntimeMoniker`

Reply:
Yes of course that's fine.